src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/VariantArray.cs

```diff
@@ -49,11 +50,15 @@ internal static class VariantArray
         // (guarenteed less than 28, in practice 0-2)
         private static readonly List<Type> s_generatedTypes = new List<Type>(0);
 
+        [RequiresUnreferencedCode("Types and members might be removed")]
```

Comment:
This could be made safe by having a `DynamicDependency` for the above `VariantArray1,2,4,8` types and then suppressing the warning.

The expression would either be for one of the above types, or a dynamically generated type, which isn't subject to trimming.